compiler/modules/bir/verify.bal

```diff
@@ -67,18 +67,65 @@ class VerifyContext {
 
 public function verifyFunctionCode(Module mod, FunctionDefn defn, FunctionCode code) returns Error? {
     VerifyContext cx = new(mod, defn);
+    boolean[] tmpRegisterUsed = [];
+    if code.registers.length() > 0 {
+        tmpRegisterUsed[code.registers.length() - 1] = false;
+    }
     foreach BasicBlock b in code.blocks {
-        check verifyBasicBlock(cx, b);
+        check verifyBasicBlock(cx, b, tmpRegisterUsed);
     }
 }
 
 type IntBinaryInsn IntArithmeticBinaryInsn|IntBitwiseBinaryInsn;
 
 type Error err:Semantic|err:Internal;
 
-function verifyBasicBlock(VerifyContext vc, BasicBlock bb) returns Error? {
+function verifyBasicBlock(VerifyContext vc, BasicBlock bb, boolean[] tmpRegisterUsed) returns Error? {
     foreach Insn insn in bb.insns {
         check verifyInsn(vc, insn);
+        check verifyInsnRegisterKinds(vc, insn, tmpRegisterUsed);
+    }
+}
+
+function verifyInsnRegisterKinds(VerifyContext vc, Insn insn, boolean[] tmpRegisterUsed) returns Error? {
+    if insn is ResultInsnBase {
+        if tmpRegisterUsed[insn.result.number] {
+            return vc.invalidErr("tmp register defined in multiple places", insn.pos);
+        }
+        else {
+            tmpRegisterUsed[insn.result.number] = true;
+        }
+    }
+
+    match insn {
+        var { operand } => {
+            check verifyRegisterKind(vc, operand, tmpRegisterUsed, insn.pos);
+        }
+        var { operands } => {
```

Comment:
Can we do `var { operands } | { args: operands } => {` not sure if it works in jBallerina, if not I think a issue needs to be created.

Anyway I think `CallInsn` needs to be changed  to have `operands` instead of `args` for constancy. Please also create an issue for that.

Reply:
We can do `var { operands } | var { args: operands } => {`